src/main/java/org/variantsync/diffdetective/analysis/LineGraphExportAnalysis.java

```diff
@@ -0,0 +1,91 @@
+package org.variantsync.diffdetective.analysis;
+
+import java.io.OutputStream;
+import java.util.LinkedHashMap;
+
+import org.variantsync.diffdetective.analysis.AnalysisResult.ResultKey;
+import org.variantsync.diffdetective.analysis.strategies.AnalysisStrategy;
+import org.variantsync.diffdetective.metadata.Metadata;
+import org.variantsync.diffdetective.variation.diff.serialize.LineGraphExport;
+import org.variantsync.diffdetective.variation.diff.serialize.LineGraphExportOptions;
+import org.variantsync.functjonal.category.InplaceSemigroup;
+
+public class LineGraphExportAnalysis implements Analysis.Hooks {
+    public static final ResultKey<Result> RESULT = new ResultKey<>("LineGraphExportAnalysis");
+    public static final class Result implements Metadata<Result> {
+        public String treeFormat;
+        public String nodeFormat;
+        public String edgeFormat;
+
+        public static final InplaceSemigroup<Result> ISEMIGROUP = (a, b) -> {
+
+        };
```

Comment:
Throw an exception if the formats of a and b are different? Only equal formats should be merged right?

Reply:
Yes, something like
```java
if (b.treeFormat != null) {
    if (a.treeFormat != null) {
        Assert.assertTrue(a.treeFormat.equals(b.treeFormat));
    } else {
        a.treeFormat = b.treeFormat;
    }
}
```
would work. In this case we should check every semigroup instance for similar cases. Of the top of my head, I know at least one other case which requires different behavior: `AnalysisResult.repoName` can't throw an exception in such a case as it will be different in `Analysis.forEachCommit`.

Reply:
Integrated into #65, for all fields noted above.